11-integrating-with-function-calling/typescript/function-app/src/main.ts

```diff
@@ -0,0 +1,67 @@
+import { OpenAIClient, AzureKeyCredential } from "@azure/openai";
+import axios from "axios";
+import * as dotenv from "dotenv";
+
+dotenv.config();
+
+const endpoint = process.env.AZURE_OPENAI_ENDPOINT || "";
+const azureApiKey = process.env.AZURE_OPENAI_KEY || "";
+const bingMapsBaseUrl = process.env.BING_MAPS_BASE_URL || "";
+const bingApiKey = process.env.BING_API_KEY || "";
+
+async function findWeather(currentLocation: string, placeType: string) {
+  const url = `${bingMapsBaseUrl}?query=${encodeURIComponent(currentLocation)}&type=${placeType}&key=${bingApiKey}`;
+
+  try {
+    const response = await axios.get(url);
+    return response.data;
+  } catch (error) {
+    console.log('Error in findWeather...: ', error);
+  }
+}
+
+const getCurrentWeatherFunction = {
+  name: "get_current_weather",
+  description: "Get the current weather in a given location",
+  parameters: {
+    type: "object",
+    properties: {
+      location: {
+        type: "string",
+        description: "The city and state, e.g. San Francisco, CA"
+      },
+      unit: {
+        type: "string",
+        enum: ["C", "F"], // Celsius or Fahrenheit
+      },
+    },
+    required: ["location"],
+  },
+};
+
+async function main() {
+  try {
+    console.log("== Chat Completions App with Functions ==");
+
+    const client = new OpenAIClient(endpoint, new AzureKeyCredential(azureApiKey));
+    const deploymentName = "gpt-4"; // if you want to use 'gpt-4' model you need to create a resource in Sweden Central region
+    const result = await client.getChatCompletions(deploymentName, [
+      { role: "user", content: "What's the weather in New York, NYC?" }
+    ], {
+      functions: [getCurrentWeatherFunction],
+    });
+
+    for (const choice of result.choices) {
+      console.log(choice.message?.functionCall);
+
+      if (choice.message?.functionCall) {
+        let response = await findWeather("New York", "C"); // include here the city and type [C = Celsius, F = Fahrenheit]
```

Comment:
@glaucia86 should we add code how to parse out the parameters, thinking that's of value to see that the LLM does parse out parameters from a user's prompt and returns something like this JSON when `getChatCompletions`  iscalled:

```json
{
  "location": "New York",
  "unit": "F"
}
````

thoughts?

Reply:
Hi, @softchris! I loved the insight! I will change now